### ISSUE TITLE
BN in linear classification stage of simclr

### ISSUE THREAD

**Issue by haohang96** [open-mmlab/mmselfsup#11]
Based on your implementation, I found that bn(affine=false) is used when ft with simclr. But it seems that bn is not used in their official tf implementation. 

I also re-implement simclr with the same mlp architecture (refer your LARS optimizer and use apex sync bn method).  I found that the accuracy is much lower without bn in linear-classification stage. If bn is necessary based pytorch implementation?

**Comment by XiaohangZhan** [comment]
We followed "Scaling and Benchmarking Self-Supervised Visual Representation Learning" in imagenet linear evaluation, including the BN(affine=False), since this paper is earlier than SimCLR and already set up comprehensive benchmarks. For a fair comparison, we keep using this setting to evaluate SimCLR and other methods. We did not study if  BN in the linear head have any impacts. Our reproduced result is 61.6, close to the official result 61.9.
Regarding LARS, previously I confirmed with the author of SimCLR, and they actually used SGD rather than LARS in linear classification. You may also refer to `https://github.com/google-research/simclr#finetuning-the-linear-head-linear-eval` where the optimizer is `momentum` (equivalent to SGD with momentum in PyTorch), the batch size is 512, not very large actually.
You may also test your re-implemented simclr model on our SVM benchmark to see if it was corrected reproduced. Our reproduced SimCLR  (batch size 256, epoch 200) gives 78.95.

**Comment by haohang96** [comment]
Thanks for your reply!
I have another two questions: 

1) openselfsup/models/backbones/resnet.py, there is a function 'train(self, mode=True)'. I wondered the purpose of this function. It seems that this function is used in validated_hooks.py to set eval model of bn. Why not just use model.eval() to achieve the same purpose? Is there some other difference between this function and model.eval()?

2) what is the difference between feat5 in ImageNet (Multi) and ImageNet (Last)? 

Thanks!

**Comment by XiaohangZhan** [comment]
After validation, you need to switch the mode to train. Using model.train() will switch all layers to training mode. However, for linear evaluation, the backbone is fixed, including the BN layers. The overwritten train() supports to keep the backbone freeze.
Following previous works, feat5 is downsampled to around 9000 dimension, while the last layer is 2048 dimensional after avgpool for resnet50.
